File: bench/benches/deforest.java
// sum of squares of the odd numbers in [0, N) -- a map/filter/fold pipeline.
// checksum = 1333333330000.
class Main {
    static long work() {
        long s = 0;
        for (int x = 0; x < 20000; x++) {
            if (x % 2 == 1) {
                s += (long) x * x;
            }
        }
        return s;
    }

    public static void main(String[] a) {
        Bench.bench("deforest", Main::work);
    }
}

File: bench/benches/deforest.clj
;; map/filter/fold list pipeline: sum the squares of the odd numbers in [0, N).
;; the lazy range + intermediate seqs are realized INSIDE the timed work. checksum = 1333333330000.
(load-file "lib/bench.clj")
(bench "deforest" (fn [] (reduce + (map #(* % %) (filter odd? (range 0 20000))))))
